SU2_CFD/src/integration/CIntegrationFactory.cpp

```diff
@@ -0,0 +1,183 @@
+/*!
+ * \file CIntegrationFactory.cpp
+ * \brief Main subroutines for CIntegrationFactory .
+ * \author T. Albring
+ * \version 7.0.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../include/integration/CIntegrationFactory.hpp"
+#include "../../include/integration/CSingleGridIntegration.hpp"
+#include "../../include/integration/CMultiGridIntegration.hpp"
+#include "../../include/integration/CStructuralIntegration.hpp"
+#include "../../include/integration/CFEM_DG_Integration.hpp"
+
+CIntegration** CIntegrationFactory::createIntegrationContainer(ENUM_MAIN_SOLVER kindMainSolver, CConfig *config){
+
+  CIntegration **integration = new CIntegration* [MAX_SOLS]();
+
+  switch (kindMainSolver) {
+    case TEMPLATE_SOLVER:
+      integration[FLOW_SOL] = createIntegration(SUB_SOLVER::TEMPLATE, config);
+      break;
+    case INC_EULER:
+      integration[FLOW_SOL] = createIntegration(SUB_SOLVER::INC_EULER, config);
```

Comment:
This looks nice @talbring :+1:, but, what if, you create an enum for the type of integration instead and add a method to CSolver that returns the required type of integration? This factory would then just be a loop over the non-null entries of the solver container with a switch for the type of integration.

Reply:
Yep good idea! I did something similar. Instead of putting it into the solver, the solver factory now creates some meta data for every solver. So far it only stores the type of the solver and the integration method, but the ultimate goal would be that it stores also the kind of numerics that should be allocated and so on. This way there is only one place to edit in case someone adds a new solver.